src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -455,6 +457,8 @@ void DeleteInvalidBlocks(Keccak invalidBlockHash)
                     options,
                     new GethLikeBlockTracer(GethTraceOptions.Default),
                     DumpOptions.Geth);
+                
+                OnInvalidBlock?.Invoke(this, ex);
```

Comment:
I think this should go before tracing, so be first in the catch block.

Reply:
Ok